addons/account_edi_ubl_cii/models/account_edi_xml_ubl_20.py

```diff
@@ -521,7 +521,7 @@ def _export_invoice_vals(self, invoice):
         sales_order_id = 'sale_line_ids' in invoice.invoice_line_ids._fields \
                          and ",".join(invoice.invoice_line_ids.sale_line_ids.order_id.mapped('name'))
         # OrderReference/ID (order_reference) is mandatory inside the OrderReference node !
-        order_reference = invoice.ref or invoice.name
+        order_reference = invoice.ref or invoice.name or invoice.name_placeholder_form
```

Comment:
Why wouldn't we have a name? Why would it be draft?
Should we really give a name if it's not the right one?


____
*Probably not relevant:
With the previous mechanism, we would have been more confident that the number would stick because the next invoice would be with sequence 2, so no conflict possible (whereas with `name_placeholder_form` you have duplicates when in draft)*

Reply:
For the name: as i said in another comment, when there is no posted move in the sequence, all names are False so the placeholder is shown. Once a move gets posted, all names should be '/' so "Draft" is displayed in the form view. Depending on whether we change that, it will be adapted

For the draft: I'm not aware of all the systems i'm adapting so i try to change the flows as little as possible but i'm often wondering whether we should not limit specific flows to posted moves. Here, i adapt the `order_reference` value to make sure it is set to something when the move is in draft but seeing this is a method to export an invoice, it could be also solved by just making sure the move is posted (and therefore has a name). I just thought if it was made possible it was probably for a reason and shouldn't change...

And so all in all, i have no choice but to give it a potential wrong order_ref if we keep the possibility to use that method on 'draft' moves. Wyt ?